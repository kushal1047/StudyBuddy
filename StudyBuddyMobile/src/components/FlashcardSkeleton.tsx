import React from "react";
import { View } from "react-native";
import ContentLoader, { Rect } from "react-content-loader/native";

export default function FlashcardSkeleton() {
  return (
    <View
      style={{
        backgroundColor: "white",
        borderRadius: 12,
        padding: 16,
        marginBottom: 12,
        shadowColor: "#000",
        shadowOffset: { width: 0, height: 1 },
        shadowOpacity: 0.1,
        shadowRadius: 4,
        elevation: 2,
      }}
    >
      <ContentLoader
        speed={1}
        width={350}
        height={120}
        backgroundColor="#f3f3f3"
        foregroundColor="#ecebeb"
      >
        <Rect x="0" y="0" rx="4" ry="4" width="80" height="12" />
        <Rect x="0" y="20" rx="4" ry="4" width="300" height="16" />
        <Rect x="0" y="50" rx="4" ry="4" width="80" height="12" />
        <Rect x="0" y="70" rx="4" ry="4" width="250" height="16" />
      </ContentLoader>
    </View>
  );
}
