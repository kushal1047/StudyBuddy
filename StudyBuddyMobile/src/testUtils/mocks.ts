// Sample user for testing
export const mockUser = {
  id: 1,
  username: "testuser",
  email: "test@example.com",
  createdAt: "2024-01-01T00:00:00Z",
};

// Sample deck for testing
export const mockDeck = {
  id: 1,
  title: "Test Deck",
  description: "Test Description",
  createdAt: "2024-01-01T00:00:00Z",
  updatedAt: "2024-01-01T00:00:00Z",
  userId: 1,
  flashcardCount: 5,
};

// Sample flashcard for testing
export const mockFlashcard = {
  id: 1,
  question: "What is React?",
  answer: "A JavaScript library",
  createdAt: "2024-01-01T00:00:00Z",
  updatedAt: "2024-01-01T00:00:00Z",
  deckId: 1,
  deckTitle: "Test Deck",
};

// Sample stats for testing
export const mockStatistics = {
  totalDecks: 5,
  totalFlashcards: 25,
  totalStudySessions: 50,
  correctAnswers: 35,
  incorrectAnswers: 15,
  accuracyPercentage: 70.0,
  lastStudyDate: "2024-01-01T00:00:00Z",
};
