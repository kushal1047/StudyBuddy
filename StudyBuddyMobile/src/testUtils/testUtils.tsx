import React from "react";
import { render, RenderOptions } from "@testing-library/react-native";

// Mock providers wrapper
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <>{children}</>;
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react-native";
export { customRender as render };
