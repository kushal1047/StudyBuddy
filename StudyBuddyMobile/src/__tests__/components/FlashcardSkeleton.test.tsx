import React from "react";
import { render } from "../../testUtils/testUtils";
import FlashcardSkeleton from "../../components/FlashcardSkeleton";

describe("FlashcardSkeleton", () => {
  it("should render without crashing", () => {
    const { toJSON } = render(<FlashcardSkeleton />);
    expect(toJSON()).toBeTruthy();
  });

  it("should match snapshot", () => {
    const { toJSON } = render(<FlashcardSkeleton />);
    expect(toJSON()).toMatchSnapshot();
  });
});
