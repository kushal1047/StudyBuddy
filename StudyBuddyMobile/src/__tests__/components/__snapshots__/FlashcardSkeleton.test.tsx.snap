// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashcardSkeleton should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 12,
      "elevation": 2,
      "marginBottom": 12,
      "padding": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 4,
    }
  }
>
  <RNSVGSvgView
    bbHeight={120}
    bbWidth={350}
    focusable={false}
    height={120}
    interval={0.25}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {},
        {
          "flex": 0,
          "height": 120,
          "width": 350,
        },
      ]
    }
    width={350}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGRect
        clipPath="gq7zcnp-animated-diff"
        fill={
          {
            "brushRef": "gq7zcnp-diff",
            "type": 1,
          }
        }
        height="100%"
        propList={
          [
            "fill",
          ]
        }
        width="100%"
        x="0"
        y="0"
      />
      <RNSVGDefs>
        <RNSVGClipPath
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          name="gq7zcnp-animated-diff"
        >
          <RNSVGRect
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            height="12"
            rx="4"
            ry="4"
            width="80"
            x="0"
            y="0"
          />
          <RNSVGRect
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            height="16"
            rx="4"
            ry="4"
            width="300"
            x="0"
            y="20"
          />
          <RNSVGRect
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            height="12"
            rx="4"
            ry="4"
            width="80"
            x="0"
            y="50"
          />
          <RNSVGRect
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            height="16"
            rx="4"
            ry="4"
            width="250"
            x="0"
            y="70"
          />
        </RNSVGClipPath>
        <RNSVGLinearGradient
          gradient={
            [
              0,
              -789517,
              0.5,
              -1250325,
              1,
              -789517,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="gq7zcnp-diff"
          x1="-100%"
          x2="0%"
          y1={0}
          y2={0}
        />
      </RNSVGDefs>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
